lib/modules/datasource/deno/index.ts

```diff
@@ -0,0 +1,137 @@
+import is from '@sindresorhus/is';
+import pMap from 'p-map';
+import { logger } from '../../../logger';
+import * as packageCache from '../../../util/cache/package';
+import { cache } from '../../../util/cache/package/decorator';
+import { regEx } from '../../../util/regex';
+import { joinUrlParts } from '../../../util/url';
+import * as semanticVersioning from '../../versioning/semver';
+import { Datasource } from '../datasource';
+import type { Release } from '../index';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type {
+  DenoAPIModuleResponse,
+  DenoAPIModuleVersionResponse,
+  ReleaseMap,
+} from './types';
+import { createSourceURL, tagsToRecord } from './utils';
+
+export class DenoDatasource extends Datasource {
+  static readonly id = 'deno';
+
+  override readonly customRegistrySupport = true;
+
+  override readonly registryStrategy = 'first';
+
+  override readonly defaultVersioning = semanticVersioning.id;
+
+  override readonly defaultRegistryUrls = ['https://apiland.deno.dev'];
+
+  constructor() {
+    super(DenoDatasource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${DenoDatasource.id}`,
+    key: ({ packageName, registryUrl }: GetReleasesConfig) =>
+      // TODO: types (#7154)
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      `${registryUrl}:${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const massagedRegistryUrl = registryUrl ?? this.defaultRegistryUrls[0];
```

Comment:
```suggestion
    const massagedRegistryUrl = registryUrl!;
```
will never null or undefined, the upper layer will enforce this. so it's just a typescript issue.